cmd/root.go

```
@@ -0,0 +1,154 @@
+func Execute(version, commit string) {
+	gVersion = version
+	gCommit = commit
+	if err := RootCmd.Execute(); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
+
+func init() {
+	convertCmd.PersistentFlags().StringVarP(&gConfigFilePath, "config", "c", "", "specify an alternate config file [env KROKI_CONFIG]")
+	convertCmd.PersistentFlags().StringVarP(&gDiagramType, "type", "t", "", "specify the diagram type [actdiag, blockdiag, c4plantuml, ditaa, dot, erd, graphviz, nomnoml, nwdiag, plantuml, seqdiag, svgbob, umlet] (default: infer from file extension)")
+	convertCmd.PersistentFlags().StringVarP(&gImageFormat, "format", "f", string(kroki.Svg), "specify the output format (default: svg)")
+	// -o, --out-file FILE              output file (default: based on path of input file); use - to output to STDOUT
+
+	RootCmd.AddCommand(versionCmd)
+	RootCmd.AddCommand(convertCmd)
+
+	cobra.OnInitialize(buildClient)
+}
+
+func outputFilePath(fileName string, imageFormat kroki.ImageFormat) string {
+	fileExtension := filepath.Ext(fileName)
+	return strings.Replace(fileName, fileExtension, "." + string(imageFormat), 1)
+}
+
+func inferDiagramType(fileName string) kroki.GraphFormat {
+	switch fileExtension := filepath.Ext(fileName); fileExtension {
+	case ".dot", ".gv", ".graphviz":
+		return kroki.Graphviz
+	case ".puml", ".plantuml":
+		return kroki.Plantuml
+	case ".nomnoml":
+		return kroki.Nomnoml
+	case ".blockdiag":
+		return kroki.BlockDiag
+	case ".mermaid":
+		return kroki.Mermaid
+	case ".svgbob":
+		return kroki.Svgbob
+	case ".umlet":
+		return kroki.Umlet
+	case ".c4puml", ".c4":
+		return kroki.C4plantuml
+	case ".seqdiag":
+		return kroki.SeqDiag
+	case ".erd":
+		return kroki.GraphFormat("erd")
+	case ".nwdiag":
+		return kroki.GraphFormat("nwdiag")
+	case ".actdiag":
+		return kroki.GraphFormat("actdiag")
+	case ".ditaa":
+		return kroki.GraphFormat("ditaa")
+	default:
+		return kroki.GraphFormat("")
```

You should return an error is the format is not supported.

---

done